test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSuite.java

```diff
@@ -152,6 +152,15 @@ public void validate() {
                 "without a corresponding [\"skip\": \"features\": \"warnings\"] so runners that do not support the [warnings] " +
                 "section can skip the test at line [" + section.getLocation().lineNumber + "]");
 
+        errors = Stream.concat(errors, sections.stream().filter(section -> section instanceof DoSection)
```

Comment:
I wonder if all this would become more readable if it were a for loop. Its certainly faster to copy and paste the blocks but maybe its time to redo?